src/Orleans.Core/Runtime/OutsideRuntimeClient.cs

```diff
@@ -125,34 +125,28 @@ internal void ConsumeServices(IServiceProvider services)
             this.SerializationManager = this.ServiceProvider.GetRequiredService<SerializationManager>();
             this.messageFactory = this.ServiceProvider.GetService<MessageFactory>();
 
-            this.config = this.ServiceProvider.GetRequiredService<ClientConfiguration>();
-
-            var resolvedClientMessagingOptions = this.ServiceProvider.GetRequiredService<IOptions<ClientMessagingOptions>>();
-            this.clientMessagingOptions = resolvedClientMessagingOptions.Value;
-
             this.GrainReferenceRuntime = this.ServiceProvider.GetRequiredService<IGrainReferenceRuntime>();
 
             var statisticsOptions = this.ServiceProvider.GetRequiredService<IOptions<ClientStatisticsOptions>>().Value;
             StatisticsCollector.Initialize(statisticsOptions.CollectionLevel);
 
-            BufferPool.InitGlobalBufferPool(resolvedClientMessagingOptions.Value);
+            BufferPool.InitGlobalBufferPool(this.clientMessagingOptions);
 
             try
             {
                 AppDomain.CurrentDomain.DomainUnload += CurrentDomain_DomainUnload;
 
                 clientProviderRuntime = this.ServiceProvider.GetRequiredService<ClientProviderRuntime>();
 
-                responseTimeout = Debugger.IsAttached ? Constants.DEFAULT_RESPONSE_TIMEOUT : config.ResponseTimeout;
-                this.localAddress = ConfigUtilities.GetLocalIPAddress(config.PreferredFamily, config.NetInterface);
+                responseTimeout = Debugger.IsAttached ? Constants.DEFAULT_RESPONSE_TIMEOUT : this.clientMessagingOptions.ResponseTimeout;
+                this.localAddress = ConfigUtilities.GetLocalIPAddress(this.clientMessagingOptions.PreferredFamily, this.clientMessagingOptions.NetInterface);
 
                 // Client init / sign-on message
                 logger.Info(ErrorCode.ClientInitializing, string.Format(
                     "{0} Initializing OutsideRuntimeClient on {1} at {2} Client Id = {3} {0}",
-                    BARS, config.DNSHostName, localAddress, handshakeClientId));
+                    BARS, Dns.GetHostName(), localAddress, handshakeClientId));
```

Comment:
Is there a reason why we might want hostname to be configurable on the client? I'm not sure, but this is ok for now

Reply:
the xml on `ClientConfiguration.DNSHostName` is 

/// <summary>Gets the true host name, no IP address. It equals Dns.GetHostName()</summary>

So I think this change is safe.